gabarit/template_nlp/nlp_project/package_name/utils.py

```diff
@@ -192,6 +192,36 @@ def wrapper(self, x, *args, **kwargs):
     return wrapper
 
 
+def is_ndarray_convertable(obj: Union[np.ndarray, Any]) -> bool:
+    '''Returns True if the object is covertable to a builtin type in the same way a np.ndarray is
+    
+    Args:
+        obj (Union[np.ndarray, Any]): an object to test
+    Returns:
+        bool: True if the object is covertable to a list as a np.ndarray is
+    '''
+    return hasattr(obj, "dtype") and hasattr(obj, "astype") and hasattr(obj, "tolist")
+
+
+def ndarray_to_builtin_object(obj: Any) -> Any:
+    '''Transform a numpy.ndarray like object to a builtin type like int, float or list
+
+    Args:
+        obj (Any): An object
+    Returns:
+        Any: The object converted to a builtin type like int, float or list
+    '''
+    if is_ndarray_convertable(obj):
```

Comment:
Maybe we should trigger an error here if not convertable ?
It can be misleading to call `ndarray_to_builtin_object` and still not have a builtin object